src/frontend/components/StudentLiveControlBar/LeaveJoinDiscussionWrapper/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect } from 'react';
+import { normalizeColor } from 'grommet/utils';
+import toast from 'react-hot-toast';
+import { defineMessages, useIntl } from 'react-intl';
+import { Redirect, useLocation } from 'react-router-dom';
+
+import { useParticipantWorkflow } from 'data/stores/useParticipantWorkflow';
+import { ButtonLayout } from 'components/Button/ButtonLayout';
+import { PUBLIC_JITSI_ROUTE } from 'components/PublicVideoLiveJitsi/route';
+import { PLAYER_ROUTE } from 'components/routes';
+import { StudentJoinDiscussionButton } from 'components/StudentLiveControlButtons/StudentJoinDiscussionButton';
+import { StudentLeaveDiscussionButton } from 'components/StudentLiveControlButtons/StudentLeaveDiscussionButton';
+import { WaitingJoinDiscussionSVG } from 'components/SVGIcons/WaitingJoinDiscussionSVG';
+import { modelName } from 'types/models';
+import { theme } from 'utils/theme/theme';
+
+const messages = defineMessages({
+  waitingInstructor: {
+    defaultMessage: "Waiting for Instructor's response",
+    description: `Text that replace the JoinDiscussion button before the instructor response.`,
+    id: 'components.StudentJoinDiscussionButton.waitingInstructor',
+  },
+  rejected: {
+    defaultMessage:
+      'Your request to join the discussion has not been accepted.',
+    description: 'Text to tell participant his/her request is not accepted',
+    id: 'components.StudentJoinDiscussionButton.rejected',
+  },
+});
+
+export const LeaveJoinDiscussionWrapper = () => {
+  const intl = useIntl();
+  const location = useLocation();
+  const { asked, accepted, rejected, reset } = useParticipantWorkflow(
+    (state) => ({
+      asked: state.asked,
+      accepted: state.accepted,
+      rejected: state.rejected,
+      reset: state.reset,
+    }),
+  );
+
+  //  if user is rejected : alert user and reset state
+  useEffect(() => {
+    if (rejected) {
+      reset();
+      toast.error(intl.formatMessage(messages.rejected));
+    }
+  }, [rejected, reset]);
```

Comment:
```suggestion
  }, [rejected]);
```

as `reset` is a function ans not a value I don't think you need to pass.

Reply:
I kinda disagree here, if I don't had the function in dependencies it's value may be miss references in the closure.
In fact, this can not happen with the current implementation of zustand but this is highly depending on zustand changes (or in our implementation of the store)

Reply:
Ok you can keep it even if its reference will never change.